collectors/osd.go

```diff
@@ -750,29 +816,51 @@ func (o *OSDCollector) collectOSDScrubState(ch chan<- prometheus.Metric) error {
 
 func (o *OSDCollector) collectPGRecoveryState(ch chan<- prometheus.Metric) error {
 	for _, pg := range o.pgDumpBrief {
-		if o.initialCollect {
+
+		// We need previous PG state in order to update the metric when a PG has
+		// completed recovery or backfill. Or it could be an empty string if
+		// unknown.
+		prevPGState := o.pgStateCache[pg.PGID]
```

Comment:
Would be nice to do `prevPGState, okPrevPGState := o.pgStateCache[pg.PGID]` and then check for validity of `okPrevPGState` instead of checking for empty string.